sale_stock_release_channel_partner_by_date/models/sale_order.py

```diff
@@ -42,7 +42,8 @@ def _compute_release_channel_id(self):
             if not rec._check_release_channel_partner_date_requirements():
                 continue
             channel_date = rec.release_channel_partner_date_id
-            rec.release_channel_id = channel_date.release_channel_id
+            if channel_date:
```

Comment:
Yes indeed. But I have still one open questions about this (you can check the test where there are FIXME in this PR):

If we have configured a channel **X** (with a carrier) and **Y** (no carrier), and have a specific channel date set for **X**.

Currently if we do not define a carrier on the SO, the computed release channel wil be taken from the configured specific channel (as it's the only one that exists). So the SO will show the release channel **X**.

Then we validate the SO, and we trigger the channel assignment on the delivery: it is now assigned to the release channel **Y** (of course, there was no carrier defined, I guess it fallbacks on **Y** because of that).

IMO this is not what the user will expect.

How should we handle SO without carrier (same question for SO without warehouse) regarding release channels?

Reply:
I agree (I don't remember why we searched on carrier_id=False in the initial implementation, we did that together and on purpose :thinking: ).

Reply:
For the warehouse, while we could state it is always set on the SO (I could not find some without WH in customers DBs), it is however not mandatory on the release channel. We should consider these release channels as elligible, and thus keep our check on warehouse_id=False in the domain.